### PR TITLE
enable config tls for pulsar

### PR DESCRIPTION
**Pull request by HaitaoDeng**
image

### CODE REVIEW

File: lib/api/pulsar/pulsar_tenant_api.dart [hunk 1]
```diff
@@ -20,53 +20,52 @@
...
 import 'dart:developer';
 
-import 'package:http/http.dart' as http;
 import 'package:paas_dashboard_flutter/api/http_util.dart';
+import 'package:paas_dashboard_flutter/api/tls_context.dart';
 import 'package:paas_dashboard_flutter/module/pulsar/pulsar_tenant.dart';
 
 class PulsarTenantApi {
-  static Future<void> createTenant(String host, int port, String tenant) async {
+  static Future<void> createTenant(int id, String host, int port, String tenant, TlsContext tlsContext) async {
```
Comment (Shoothzj) [inline lib/api/pulsar/pulsar_tenant_api.dart:28]:
```suggestion
  static Future<void> createTenant(int id, String host, int port, TlsContext tlsContext, String tenant) async {
```

Comment (HaitaoDeng) [reply lib/api/pulsar/pulsar_tenant_api.dart]: OK

File: lib/persistent/persistent_db.dart [hunk 2]
```diff
@@ -132,11 +134,70 @@ class PersistentDb implements PersistentApi {
...
+      String functionHost,
+      int functionPort,
+      bool enableTls,
+      bool functionEnableTls,
+      String caFile,
+      String clientCertFile,
+      String clientKeyFile,
+      String clientKeyPassword) async {
+    var aux = await getInstance();
+    // var map = {"name": name, "host": host, "port": port, "function_host": functionHost, "function_port": functionPort, "enable_tls" enableTls ? TlsContext.ENABLE_TLS : TlsContext.DIS_ENABLE_TLS, "ca_file": caFile, "client_cert_file": clientCertFile, "client_key_file": clientKeyFile, "client_key_password": clientKeyPassword};
```
Comment (lovehzj) [inline lib/persistent/persistent_db.dart:183]: comment out code

File: lib/api/tls_context.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+class TlsContext {
+  static const int ENABLE_TLS = 1;
+  static const int DIS_ENABLE_TLS = 0;
```
Comment (lovehzj) [inline lib/api/tls_context.dart:22]:
```suggestion
  static const int DISABLE_TLS = 0;
```
why don't use boolean variable?

File: lib/api/http_util.dart [hunk 4]
```diff
@@ -26,3 +73,8 @@ class HttpUtil {
     return code >= 200 && code < 300;
   }
 }
+
+enum SERVER {
```
Comment (lovehzj) [inline lib/api/http_util.dart:77]: should this enum put in module package?

File: lib/persistent/persistent_db.dart (hunk 2 above)
Comment (HaitaoDeng) [reply lib/persistent/persistent_db.dart:183]: OK

File: lib/api/tls_context.dart (hunk 3 above)
Comment (HaitaoDeng) [reply lib/api/tls_context.dart:22]: image
used bool, but sqllite don't support bool, Persistent use 0 or 1

File: lib/api/http_util.dart (hunk 4 above)
Comment (HaitaoDeng) [reply lib/api/http_util.dart:77]: I think put here, SERVER current only used in httpUtil

**Review by goflutterjava — APPROVED** [review]

**Pull request merged**